src/platform/i386/runscripts/unit_hierschedcomps.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+cp llboot_comp.o llboot.o
+cp hier_fprr.o 3_2_hier_fprr.o
+cp hier_fprr.o 4_2_hier_fprr.o
+cp hier_fprr.o 5_3_hier_fprr.o
+cp unit_schedcomp_test.o _2_unit_schedcomp_test.o
+cp unit_schedcomp_test.o _3_unit_schedcomp_test.o
+cp unit_schedcomp_test.o _4_unit_schedcomp_test.o
+cp unit_schedcomp_test.o _5_unit_schedcomp_test.o
+./cos_linker "llboot.o, ;resmgr.o, ;root_fprr.o, ;3_2_hier_fprr.o, ;4_2_hier_fprr.o, ;5_3_hier_fprr.o, ;_2_unit_schedcomp_test.o, ;_3_unit_schedcomp_test.o, ;_4_unit_schedcomp_test.o, ;_5_unit_schedcomp_test.o, :root_fprr.o-resmgr.o;3_2_hier_fprr.o-resmgr.o|[parent_]root_fprr.o;4_2_hier_fprr.o-resmgr.o|[parent_]root_fprr.o;5_3_hier_fprr.o-resmgr.o|[parent_]3_2_hier_fprr.o;_2_unit_schedcomp_test.o-root_fprr.o;_3_unit_schedcomp_test.o-3_2_hier_fprr.o;_4_unit_schedcomp_test.o-4_2_hier_fprr.o;_5_unit_schedcomp_test.o-5_3_hier_fprr.o" ./gen_client_stub
```

Comment:
Wow.. Got 9 components booting with 3 levels of hierarchy! This although is painful to get right, feels so much more easier than getting fork() version to boot as many components with multiple levels of hierarchy! :P 

Reference of a working output on Qemu!!
```
[21:15:41] transfer$ ./qemu.sh unit_hierschedcomps.sh 
System memory info from multiboot (end 0xc2524000):
	Modules:
	- 0: [02525000, 0264b000) : 40800000-40802fc0 @ virtual address 40800000, _start = 40802fc0.
	Memory regions:
	- 0 (Available): [00000000, 0009fc00)
	- 1 (Reserved ): [0009fc00, 000a0000)
	- 2 (Reserved ): [000f0000, 00100000)
	- 3 (Available): [00100000, 2fffe000)
	  memory available at boot time: 2d9b3000 (729 MB + 716 KB)
	- 4 (Reserved ): [2fffe000, 30000000)
	- 5 (Reserved ): [fffc0000, 100000000)
	Amount of wasted memory due to layout is 0 MB + 0x0 B
Processor information:
	Vendor: GenuineIntel
	Features [rdtsc fpu superpages sysenter/exit apic ]
	Frequency: 0 (* 100Mz)
Initializing virtual memory
	Setting up initial page directory.
ACPI initialization
	Found good RSDP @ c00f0990
	Found good HPET @ f03ffb89
Initializing HPET @ f03ffb89
	Checksum is OK
	Addr: 00000000fed00000
	hpet: fed00000
	Set HPET @ f0500000
	Found good APIC @ f03ffb11
Initializing LAPIC @ f03ffb11
	MADT length 120 (base struct 44)
	LAPIC found: coreid 0, apicid 0
	I/O APIC found: ioapicid 0, addr fec00000, int offset 0
	Interrupt controller type 2: ignoring
	Interrupt controller type 2: ignoring
	Interrupt controller type 2: ignoring
	Interrupt controller type 2: ignoring
	Interrupt controller type 2: ignoring
	Interrupt controller type 4: ignoring
	APICs processed, 1 cores
	Checksum is OK
	lapic: fee00000
	Set LAPIC @ f0a00000
Setting up the booter component.
	Creating 1 booter VM PTEs for PGD @ 0xc264e000 from [c2525000,c264b000) to [40800000,40926000).
	Mapping in booter VM.
	Creating 183 untyped memory PTEs for PGD @ 0xc0119000 from [c2707000,efffe000) to [400000,2dcf7000).
	Mapping in untyped memory.
	Capability table and page-table created.
	Created boot component structure from page-table and capability-table.
	Creating initial threads, tcaps, and rcv end-points in boot-component.
	Boot component initialization complete.
Enabling timer @ f0500000 with tick granularity 10000 picoseconds
LAPIC: TSC-Deadline Mode not supported! Configuring Oneshot Mode!
Upcall into boot component at ip 0x40802fc0
------------------[ Kernel boot complete ]------------------
0=> Booter for new kernel
0=> MAX DEPS: 1024
0=> ndeps: 0
0=> num cobjs: 9
0=> cobj resmgr:1 found at 0x40870000:1afa7, -> 44800000
0=> cobj root_fprr:2 found at 0x4088afc0:1845b, -> 44c00000
0=> cobj 3_2_hier_fprr:3 found at 0x408a3440:1953b, -> 45000000
0=> cobj 4_2_hier_fprr:4 found at 0x408bc980:1953b, -> 45400000
0=> cobj 5_3_hier_fprr:5 found at 0x408d5ec0:1953b, -> 45800000
0=> cobj _2_unit_schedcomp_test:6 found at 0x408ef400:cf27, -> 45c00000
0=> cobj _3_unit_schedcomp_test:7 found at 0x408fc340:cf27, -> 46000000
0=> cobj _4_unit_schedcomp_test:8 found at 0x40909280:cf27, -> 46400000
0=> cobj _5_unit_schedcomp_test:9 found at 0x409161c0:cf27 -> 46800000
LAPIC: Timer calibrated - CPU Cycles to APIC Timer Ratio is 2
Timer calibrated:
	CPU cycles per HPET tick: 23
	HPET ticks in 1000 us: 100000
0=> Comp 1 (resmgr) created @ 44800000!
0=> Comp 2 (root_fprr) created @ 44c00000!
0=> Comp 3 (3_2_hier_fprr) created @ 45000000!
0=> Comp 4 (4_2_hier_fprr) created @ 45400000!
0=> Comp 5 (5_3_hier_fprr) created @ 45800000!
0=> Comp 6 (_2_unit_schedcomp_test) created @ 45c00000!
0=> Comp 7 (_3_unit_schedcomp_test) created @ 46000000!
0=> Comp 8 (_4_unit_schedcomp_test) created @ 46400000!
0=> Comp 9 (_5_unit_schedcomp_test) created @ 46800000!
0=> Component 0 => child bitmap 3, sched bitmap 2
0=> Component 1 => child bitmap 0, sched bitmap 0
0=> Component 2 => child bitmap 2c, sched bitmap c
0=> Component 3 => child bitmap 50, sched bitmap 10
0=> Component 4 => child bitmap 80, sched bitmap 0
0=> Component 5 => child bitmap 100, sched bitmap 0
0=> Component 6 => child bitmap 0, sched bitmap 0
0=> Component 7 => child bitmap 0, sched bitmap 0
0=> Component 8 => child bitmap 0, sched bitmap 0
0=> Component 9 => child bitmap 0, sched bitmap 0
0=> Booter: done creating system.
0=> ********************************
1=> CPU cycles per sec: 2300
1=> Schedulers bitmap: 1e
1=> Initialized RESOURCE MANAGER
2=> CPU cycles per sec: 2300
2=> Child bitmap: 2c, Child-sched bitmap: c
2=> Initializing child component 3
2=> Initializing child component 4
2=> Initializing child component 6
0=> Done Initializing
0=> ********************************
0=> Root scheduler is 2
0=> Starting llboot sched loop
6=> Unit-test scheduling component
2=> SELF (inc. CHILD) INIT DONE.
3=> CPU cycles per sec: 2300
4=> CPU cycles per sec: 2300
3=> Child bitmap: 50, Child-sched bitmap: 10
3=> Initializing child component 5
4=> Child bitmap: 80, Child-sched bitmap: 0
4=> Initializing child component 8
3=> Initializing child component 7
0=> Done Initializing
0=> ********************************
0=> Done Initializing
0=> ********************************
8=> Unit-test scheduling component
3=> SELF (inc. CHILD) INIT DONE.
4=> SELF (inc. CHILD) INIT DONE.
7=> Unit-test scheduling component
5=> CPU cycles per sec: 2300
5=> Child bitmap: 100, Child-sched bitmap: 0
5=> Initializing child component 9
0=> Done Initializing
0=> ********************************
9=> Unit-test scheduling component
5=> SELF (inc. CHILD) INIT DONE.
7=> Test successful! Highest was scheduled only!
8=> Test successful! Highest was scheduled only!
9=> Test successful! Highest was scheduled only!
6=> Test successful! Highest was scheduled only!
8=> Test successful! We swapped back and forth!
8=> Done testing, spinning...
Serial: 97
9=> Test successful! We swapped back and forth!
9=> Done testing, spinning...
7=> Test successful! We swapped back and forth!
7=> Done testing, spinning...
6=> Test successful! We swapped back and forth!
6=> Done testing, spinning...
QEMU: Terminated
[21:16:39] transfer$ 

```